napari/layers/image/image.py

```diff
@@ -692,6 +698,43 @@ def _set_view_slice(self):
             self.reset_contrast_limits()
             self._should_calc_clims = False
 
+    def _project_slice(self):
+        """
+        Project a thick nD slice into a single 2d or 3d image that can be passed to the GPU.
+
+        Projection is based on slice thickness and the current projection mode.
+        """
+        if self.slice_projection == ImageSliceProjection.NONE:
+            return self.data[self._slice_indices]
+
+        half_thickness = (
+            np.maximum(np.ceil(self._thickness).astype(int), 1) / 2
+        )
+        image_slices = tuple(
+            slice(
+                ceil(max(0, self._slice_indices[i] - half_thickness[i])),
+                ceil(
+                    min(
+                        self._slice_indices[i] + half_thickness[i],
+                        self.data.shape[i],
+                    )
+                ),
+            )
+            if i in self._dims_not_displayed
+            else self._slice_indices[i]
+            for i in range(self.ndim)
+        )
+
+        proj_funcs = {
+            ImageSliceProjection.ADDITIVE: np.sum,
+            ImageSliceProjection.AVERAGE: np.mean,
+        }
+
+        image = proj_funcs[self.slice_projection](
```

Comment:
This will trigger compute with dask etc - i think it's ok that projection doesn't work with async or multiscale right now - but @perlman might want to look at this too

Reply:
I assume this would work fine with the dask job graph and get computed once at the end. Why does this not work?

Reply:
ah - ok, i couldn't remember anymore if some of those other numpy operations triggered compute - my bad!

Reply:
I still don't entirely understand what magic lets np.sum(dask.array) return a dask.array, but it works. 👍 

I'm more worried about how this would interact with a lightweight (non-dask) data loader; but a solution for 3D would work here too [e.g., making sure data associated with a slice object are loaded].

Reply:
>  still don't entirely understand what magic lets np.sum(dask.array) return a dask.array, but it works.

I believe that is the [NEP-18 __array_function__ protocol](https://numpy.org/neps/nep-0018-array-function-protocol.html) which is implemented in Dask, CuPy and possibly a few other libraries.

Reply:
Actually, for ufuncs it is the earlier [NEP-13](https://numpy.org/neps/nep-0013-ufunc-overrides.html) protocol

Reply:
This is also coming soon to dask - might be of interest to folks https://github.com/dask/dask/pull/8750 (we should probably share more broadly within napari community)